### ISSUE TITLE
UX: ":edit *.js" should suggest :next

### ISSUE THREAD

**Issue by ChrisPenner** [neovim/neovim#12139]
### Actual behaviour
When simply trying to edit some additional files I often type `:e *.js` or something similar; this almost ALWAYS results in `E77: Too many file names`.

### Expected behaviour

It's pretty clear what the user wants in this situation; they want to open the files as buffers! I'm sure there are legacy reasons for this error, but it's a pretty poor experience these days that probably doesn't need to stick around. (at least with `set hidden` enabled)

I've learned I can run `:next *.js` which has the behaviour I want, but `next` is a pretty unintuitive name and I don't really see a reason to have more than one command for this functionality.

In the worst case, perhaps we could simply add a note saying `Maybe you meant to use :next` to the error message? It took a bit of digging to find that solution and I'd love it if others didn't need to do the same.

Thanks for all your work everyone!

**Comment by justinmk** [comment]
> perhaps we could simply add a note saying `Maybe you meant to use :next` to the error message

That seems reasonable. 

I don't think `:edit` should work like next , because then the "literal" case is not addressed[1], e.g. if you actually *want* a buffer with literaly `*` in the name.

[1] But see also:

- https://github.com/neovim/neovim/issues/37862

**Comment by armhzjz** [comment]
Hello. I want to take this issue.
Could you give me a hint on where to look to start working on this issue (i.e. a file name or module or any other reference that could bring me closer to a good start?

**Comment by justinmk** [comment]
`do_exedit()`

**Comment by armhzjz** [comment]
Thank you very much @justinmk

**Comment by pta2002** [comment]
Are you still interested in working on this @armhzjz? Since it's been 25 days and I haven't seen any [WIP] pull request. I'd be interested in doing this too if you've stopped working on the issue.

**Comment by armhzjz** [comment]
Hello @pta2002. I was on vacations and came back this Wednesday (two days ago) - that's why you saw no movement here. But I am in deed interested still on this and currently working on this. Thanks for asking.

**Comment by pta2002** [comment]
Alright, good luck!

**Comment by ritwikchakraborty123** [comment]
Hai I. Want to solve this issue

**Comment by armhzjz** [comment]
Hello @ritwikchakraborty123,
Sorry but I am currently working on it.

**Comment by Lee-yeah-sign** [comment]
okay

**Comment by keremgocen** [comment]
any updates on this one?:)

**Comment by armhzjz** [comment]
PR created. It seems it has big chances of being merged as long as I provide a test.
:)

**Comment by yunus776** [comment]
> Çok teşekkür ederim @justinmk

**Comment by nc1337** [comment]
Is this issue still open?? @armhzjz. 
If not i would like to start on this issue.

**Comment by armhzjz** [comment]
Hi @nc1337 , still on.
FYI, it was decided that this issue should be merged in vim.

**Comment by Anupam-Ashish-Minz** [comment]
Hey @armhzjz, I would like to help you on this issue can you tell me how you are proceeding on this issue. I am new here.

**Comment by ritwikchakraborty123** [comment]
This still on? I was also willing to work on this . Someone told he is working.Let me know any update

**Comment by SkohTV** [comment]
I went digging a little bit into this issue

The error message doesn't originate from the C code ([`do_exedit()`](https://github.com/neovim/neovim/blob/master/src/nvim/ex_docmd.c#L5919)) but from the fact that `edit` has the flag [`FILE1`](https://github.com/neovim/neovim/blob/master/src/nvim/ex_cmds.lua#L845) in [ex_cmds.lua:845](https://github.com/neovim/neovim/blob/master/src/nvim/ex_cmds.lua#L845)
So the function get short-circuited a lot earlier

Would it be better to allow for multiple files in the Lua **and then** error later in `do_exedit()` ?
Or can we do the erroring somewhere in the Lua ? (If so, not sure where)

Also took a look at https://github.com/neovim/neovim/pull/13009, but it doesn't feel right to change the error message
Commands like `:exit`, `:cd` or `:find` also can error with `E77: Too many file names`, we wouldn't want them to suggest `:next` instead

**Comment by zhangweize9-cyber** [comment]
When you input the `*` symbol, Neovim may have interpreted it as a regular expression.

**Comment by justinmk** [comment]
> Commands like `:exit`, `:cd` or `:find` also can error with `E77: Too many file names`, we wouldn't want them to suggest `:next` instead

True, but we can adjust the message to make sense. After all, if the user accidentally passes a glob to those commands, then a hint makes sense there too.

> Would it be better to allow for multiple files in the Lua **and then** error later in `do_exedit()` ?

Not sure why Lua is mentioned. Cmdline (`:edit`) is a "DWIM"/user interface, whereas Lua is programmatic. For programmatic purposes, glob/wildcard is already possible (`vim.glob`, etc.) and the error message UX is not very relevant. Though we have discussed a formal API (`nvim_edit`) for opening files, but that's a separate topic.

**Comment by SkohTV** [comment]
> we can adjust the message to make sense

Something along the lines of `E77: Too many file names (glob(s) not supported here)` should be broad enough
Also is it still recommended to send a PR to [vim/vim](http://github.com/vim/vim) and pull it here as `vim-patch` ? (https://github.com/neovim/neovim/pull/13009#issuecomment-742707335)

> Not sure why Lua is mentioned

Since switching `FILE1` to `FILES` in [ex_cmds.lua:845](https://github.com/neovim/neovim/blob/master/src/nvim/ex_cmds.lua#L845) disable the error message of `:edit *.js` and allow `do_exedit()` to be called, I figured there was some connection with Lua code
Maybe not, I'm not an expert in multi-languages codebases

But just changing the message seems the better solution, I agree
